src/presentation/Console.java

```diff
@@ -169,7 +169,8 @@ public static boolean promptToContinue() {
 			if (matchesAffirmative) {
 				isValid = true;
 			} else if (matchesNegative) {
-				isValid = true;
+				printEmptyLine();
+				break;
```

Comment:
This logic definitely works, and it's a "fix" in the sense that we no longer see the bug. The logic for determining what Boolean value to return and the loop inside of this function are tied together at the moment, which I think is a little confusing to the reader. On quick glance, I'm not 100% sure why your fix works (if I weren't the one that wrote the original code :P), I just have to trust you that it does work. Can you see if you can separate the logic for the loop inside of the function from the logic for which Boolean value to return and make that easier to understand when reading the function?

Reply:
Alright, this one made my brain sore.  There is still a solid chance that I am way off base with this one.  The return statements really confound me...I think that maybe it should return a boolean that indicates whether or not the user is actually intending to continue playing.  In my mind though, I guess I have trouble reasoning out having a boolean there that isn't affecting whether or not the loop is still going and calling the function.  I plan to keep looking into this, is there a solid way to google this to find relevant examples? Thanks for taking the time to assign this side project, it's really helpful!

Reply:
`I think that maybe it should return a boolean that indicates whether or not the user is actually intending to continue playing`

This is excellent insight and just what I was fishing for! We have this `isValid` Boolean going on right now, which I would assume (if I didn't know how the logic of the function worked) was intended to tell whether or not the user had input a valid response.

The goal of the function, though, is to tell the calling code (aka wherever this function is actually called, aka the application class in this case) whether or not the user wants to actually continue with the game.

I think you could figure out a way to have the `isValid` variable just be used to manage the while loop inside of the function and create a new variable to keep track of which choice the user has actually made. You could then return that new variable at the end of the function and set it's correct value in the different if/else branches inside of the function.

You're on the right track!

Reply:
Doh, just realized that you already pushed a change similar to what I was talking about above...

This doesn't work like you expect though, does it? I see a little logic error in there... You're very close though!